src/event.cc

```diff
@@ -306,17 +312,26 @@ EventScheduler::find(EVH * func, void * arg)
 EventScheduler *
 EventScheduler::GetInstance()
 {
-    return &_instance;
+    static EventScheduler instance;
+    return &instance;
 }
 
 void
 EventScheduler::schedule(const char *name, EVH * func, void *arg, double when, int weight, bool cbdata)
+{
+    AsyncCall::Pointer call = asyncCall(41, 5, name, EventDialer(func, arg, cbdata));
+    schedule(name, call, when, weight);
+}
+
+void
+EventScheduler::schedule(const char *name, const AsyncCall::Pointer &call, double when, int weight)
```

Comment:
The legacy API needs to use this parameter to give ev_event the weight parameter when scheduling EventDialer based Calls. For now b7382eb makes it an un-named parameter with defaulted value of 0.

Reply:
> The legacy API needs...

This change request is explicitly and exclusively about the new public event scheduling API (whatever it will become after other fixes) -- the API that the new/converted user code will call directly to schedule AsyncCall-based events. It is not about any internally used methods or legacy API.

Reply:
My reponse was about the new API. It is used by the old API's wrappers.

Reply:
> My reponse was about the new API. It is used by the old API's wrappers.

The legacy API needs can be correctly satisfied without sabotaging the new API, without code duplication, etc. The new stand-alone AddEvent() function does not need the weight parameter and, hence, should not have it. Both new AddEvent() and the legacy API code should call the same underlying EventScheduler::schedule() method that _does_ have the legacy weight parameter. I will implement the necessary changes if you prefer; just let me know.

Reply:
A standalone AddEvent() is code duplication of eventAdd() as well as the scheduler method and this requested to be removed parameter would still need to exist for the eventAdd() variant.

Reply:
> A standalone AddEvent() is code duplication of eventAdd() as well as the scheduler method 

The above two assertions are false:

1. The new AddEvent() API does not duplicate the legacy eventAdd() API: The two functions have a different set of arguments and are designed for different kinds of event creators.
2. Corrent AddEvent() implementation will not duplicate eventAdd() implementation. Most likely, AddEvent() will just call the scheduler, supplying the right default parameters as needed. eventAdd() will create the internal AsyncCall object before calling the scheduler.
3. Corrent AddEvent() implementation will not duplicate a correctly implemented scheduler method. AddEvent() will _call_ that method instead.

> and this requested to be removed parameter would still need to exist for the eventAdd() variant.

Very true but irrelevant to this change request.